demux/demux.c

```diff
@@ -1084,14 +1085,20 @@ static void demux_shutdown(struct demux_internal *in)
     demuxer->priv = NULL;
     in->d_thread->priv = NULL;
 
-    demux_flush(demuxer);
-    assert(in->total_bytes == 0);
+    // Intentionally leak demuxer cache to avoid long and unnecessary
+    // deallocation of possibly hundreds of thousands of cached packets when
+    // exiting the player. Note that leak checkers will complain about this, so
```

Comment:
The leak checker comment is probably better placed near `quit_fast` instead of here.

Reply:
added in both places